src/scene/Entity.cpp

```diff
@@ -16,6 +16,13 @@
 
 API_OBJECT_INSTANCE(Entity);
 
+// Helper class to combine multiple callables into one overload set for std::visit.
+template<typename... Callable>
+struct Visitor : Callable...
```

Comment:
This is a general usage helper but I see no other places than this cpp file that it would be useful right now. I'd rather not create tons of utils files unless we really need them.